src/net.h

```diff
@@ -113,6 +113,14 @@ struct CSerializedNetMsg
     std::string command;
 };
 
+enum class ConnectionType {
+    INBOUND, // peer initiated connections
+    OUTBOUND, // full relay connections (blocks, addrs, txns)
+    MANUAL, // connections to addresses added via addnode or the connect command line argument
```

Comment:
My 2 cents (feel free to ignore as I'm clearly very new to the bitcoin codebase): it feels to me that the distinction manual vs non-manual (automatic?) is orthogonal to the connection type and could be a separate enum.

IIUC `MANUAL` can be composed with either `OUTBOUND`, `FEELER`, `BLOCK_RELAY` or `ADDR_FETCH`.

Reply:
See rationale here: https://github.com/bitcoin/bitcoin/pull/19316#issuecomment-646304479 and here: https://github.com/bitcoin/bitcoin/pull/19316#issuecomment-649776062

Reply:
Thanks, I should have done my homework and read that PR first (as was clearly said in the PR description!).